src/updater.js

```diff
@@ -54,18 +52,16 @@ const initialState: OfflineState = {
   }
 };
 
-// @TODO: the typing of this is all kinds of wack
-
 const offlineUpdater = function offlineUpdater(
   state: OfflineState = initialState,
-  action: ControlAction | OfflineAction | ResultAction
+  action:
+    | OfflineStatusChangeAction
+    | OfflineScheduleRetryAction
+    | ResultAction
+    | PersistRehydrateAction
 ): OfflineState {
   // Update online/offline status
-  if (
-    action.type === OFFLINE_STATUS_CHANGED &&
-    action.payload &&
```

Comment:
Were this checks redundant?

Reply:
Since now the `action.type` itself is typed as `'Offline/STATUS_CHANGED'`
```
// constants.js

export const OFFLINE_STATUS_CHANGED: 'Offline/STATUS_CHANGED' =
  'Offline/STATUS_CHANGED';
```
there is no ambiguity and the condition can rely solely on the `action.type` value